web/src/components/vis/VolcanoPlot.vue

```diff
@@ -1,17 +1,40 @@
-<script>
+<script lang="ts">
+import {
+  PropType, defineComponent, computed, ref, watchEffect, onMounted, Ref,
+} from '@vue/composition-api';
+import { axisBottom, axisLeft } from 'd3-axis';
+import { scaleLinear } from 'd3-scale';
 import { select } from 'd3-selection';
+import useAxisPlot from './use/useAxisPlot';
 
-import { axisPlot } from './mixins/axisPlot';
+interface Row {
+  pValue: number;
+  log2FoldChange: number;
+  name: string;
+  color?: string;
+}
+
+interface TransformedRow extends Row {
+  x: number;
+  y: number;
+}
 
-export default {
-  mixins: [
-    axisPlot,
-  ],
+// hoisted constants
+const margin = {
+  top: 20,
+  right: 20,
+  bottom: 50,
+  left: 50,
+};
+const radius = 3;
+const xlabel = 'log2(Fold Change)';
+const ylabel = '-log10(p-value)';
 
+export default defineComponent({
   props: {
     rows: {
-      type: Array, // {pValue: number, log2FoldChange: number, name: string, color?: string}[]
-      required: true,
+      type: Array as PropType<Row[]>,
+      default: () => [],
```

Comment:
Was there a problem with making `rows` a required prop?

Reply:
No, fixed in https://github.com/girder/viime/pull/602/commits/8b890c887770c9a09fcaec80018901a3f408ced3